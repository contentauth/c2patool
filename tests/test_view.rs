mod common;

use common::{cmd, test_img_path};
use insta_cmd::assert_cmd_snapshot;

#[test]
fn test_view_manifest() {
    assert_cmd_snapshot!(cmd().arg("view").arg("manifest").arg(test_img_path()));
}

// TODO: https://github.com/mitsuhiko/insta-cmd/issues/15
#[test]
fn test_view_manifest_debug() {
    // assert_cmd_snapshot!(cmd()
    //     .arg("view")
    //     .arg("manifest")
    //     .arg("--debug")
    //     .arg(test_img_path()));
}

// TODO: https://github.com/mitsuhiko/insta-cmd/issues/15
#[test]
fn test_view_ingredient() {
    // assert_cmd_snapshot!(cmd().arg("view").arg("ingredient").arg(test_img_path()));
}

#[test]
fn test_view_info() {
    assert_cmd_snapshot!(cmd().arg("view").arg("info").arg(test_img_path()));
}

#[test]
fn test_view_manifest_info() {
    assert_cmd_snapshot!(cmd().arg("view").arg("tree").arg(test_img_path()));
}

// TODO: https://github.com/mitsuhiko/insta-cmd/issues/15
#[test]
fn test_view_manifest_certs() {
    // let mut binding = cmd();
    // let mut cmd = binding.arg("view").arg("certs").arg(test_img_path());
    // assert_cmd_snapshot!(cmd);
}
