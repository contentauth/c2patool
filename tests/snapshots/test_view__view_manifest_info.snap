---
source: tests/test_view.rs
info:
  program: c2patool
  args:
    - view
    - tree
    - /Users/nsalvemini/Documents/repos/c2patool/tests/fixtures/C.jpg
---
success: true
exit_code: 0
----- stdout -----
Tree View:
 Asset:C.jpg, Manifest:contentauth:urn:uuid:a621a5d8-147d-499b-95d7-2eaff1009efc
├── Assertion:c2pa.thumbnail.claim.jpeg
├── Assertion:stds.schema-org.CreativeWork
├── Assertion:c2pa.actions
└── Assertion:c2pa.hash.data


----- stderr -----
